src/coremods/core_info/core_info.cpp

```diff
@@ -99,30 +99,32 @@ class CoreModInfo : public Module
 
 	void ReadConfig(ConfigStatus& status) CXX11_OVERRIDE
 	{
-		ConfigTag* tag = ServerInstance->Config->ConfValue("admin");
-		cmdadmin.AdminName = tag->getString("name");
-		cmdadmin.AdminEmail = tag->getString("email", "null@example.com");
-		cmdadmin.AdminNick = tag->getString("nick", "admin");
-
 		// Process the escape codes in the MOTDs.
-		cmdmotd.motds.clear();
+		ConfigFileCache newmotds;
 		for (ServerConfig::ClassVector::const_iterator iter = ServerInstance->Config->Classes.begin(); iter != ServerInstance->Config->Classes.end(); ++iter)
 		{
+			ConfigTag* tag = (*iter)->config;
 			// Don't process the file if it has already been processed.
-			const std::string motd = (*iter)->config->getString("motd", "motd");
-			if (cmdmotd.motds.find(motd) != cmdmotd.motds.end())
+			const std::string motd = tag->getString("motd", "motd");
+			if (newmotds.find(motd) != newmotds.end())
 				continue;
 
 			// We can't process the file if it doesn't exist.
 			ConfigFileCache::iterator file = ServerInstance->Config->Files.find(motd);
 			if (file == ServerInstance->Config->Files.end())
-				continue;
+				throw ModuleException(InspIRCd::Format("MOTD file %s not specified in <files>, at %s", motd.c_str(), tag->getTagLocation().c_str()));
```

Comment:
This is an exception to the throw-on-invalid rule because the RFC states that `RPL_NOMOTD` should be sent instead of the MOTD if the file is missing.